content/en/docs/armory-agent/armory-agent-quick.md

```diff
@@ -5,48 +5,146 @@ description: >
+The Agent consists of a plugin to Spinnaker's Clouddriver service and a K8s deployment that connects to Clouddriver. You can review the architecture in the Armory Agent [overview]({{< ref "armory-agent" >}}).
 
 ## {{% heading "prereq" %}}
 
-* This guide is for experienced Kubernetes and Armory Enterprise users.
+* This guide assumes you're using the Armory Operator to install Spinnaker, with the Kustomize method from the [spinnaker-kustomize-patch repo](https://github.com/armory/spinnaker-kustomize-patches).
+
 * You have read the Armory Agent [overview]({{< ref "armory-agent" >}}).
-* You have a Redis instance. The Agent uses Redis to coordinate between Clouddriver replicas.
 * You have configured Clouddriver to use MySQL or PostgreSQL. See the {{< linkWithTitle "clouddriver-sql-configure.md" >}} guide for instructions.
+* You have Spinnaker installed in one cluster, and an additional cluster to serve as your target deployment cluster.
+
+### Compatibility matrix
+
+{{< include "agent/agent-compat-matrix.md" >}}
+
+### Networking requirements
 
-## Networking requirements
+Communication from the Agent to Clouddriver occurs over gRPC port 9091.
 
-Communication between Clouddriver and the Agent must be `http/2`. `http/1.1` is *not* compatible and causes communication issues between Clouddriver and the Agent.  
+### Kubernetes permissions required by the Agent
 
-## Kubernetes permissions needed by the Agent
+The Agent can use a kubeconfig file loaded as a K8s secret (This is only appropriate when the agent runs in a separate cluster from the target cluster), or as a service account in the cluster it resides in. Running Agent in the target cluster with a service account is the preferred model. 
 
-The Agent should have `ClusterRole` authorization if you need to deploy pods across your cluster or `Role` authorization if you deploy pods only to a single namespace.
+## Step 1: Agent Clouddriver Plugin Installation
 
-* If Agent is running in [Agent Mode]({{< ref "armory-agent#agent-mode" >}}), then the `ClusterRole` or `Role` is the one attached to the Kubernetes Service Account mounted by the Agent pod.
-* If Agent is running in any of the other modes, then the `ClusterRole` or `Role` is the one the `kubeconfigFile` uses to interact with the target cluster. `kubeconfigFile` is configured in `kubesvc.yml` of the Agent pod.
+This step is performed in the cluster Spinnaker service is running. You will add the Clouddriver plugin and expose it as type `LoadBalancer` on gRPC port `9091`. In step 2, the Agent will be configured to communicate with Clouddriver. _Take note to ensure the plugin version is compatible with the Spinnaker version. See the comments in the manifest_.
 
-Example configuration for deploying `Pod` manifests:
+Add this manifest to your Kustomize patches:
```

Comment:
We need to say how to add: as a base? as a resource? as a patch?

I actually preferred when all these details were hidden as they add complexity for no gain.

If you want to show how to configure the plugin without kustomize, then make a separate section. Otherwise I'd keep the kustomization that just shows the thing users can change: version and config.

Reply:
I'm going with an assumption that the user knows our kustomize repo, and kustomize. I can reword from adding to kustomize patches, to place in patchesStrategicMerge.

When the details were hidden, it was very confusing. I think we need to find the balance between who we target this for. If someone know K8s, kustomize, and our kustomize repo, they will get this right away. If they don't know those elements, they won't likely have gotten to this point anyway,